tests/test_asset.py

```diff
@@ -30,6 +34,45 @@ def test_init(asset_mock):
     assert asset_mock.asset_id == ASSET_ID
 
 
+def test_repr(auth_mock):
+    asset_info = {"id": ASSET_ID}
+    asset = Asset(auth_mock, asset_info=asset_info)
+    expected_repr = "{'id': '363f89c1-3586-4b14-9a49-03a890c3b593'}"
+    assert repr(asset) == expected_repr
+
+
+@pytest.mark.parametrize(
+    "asset_id, asset_info, expected_error",
+    [
+        (
+            None,
+            None,
+            "Either asset_id or asset_info should be provided in the constructor.",
+        ),
+        (
+            "some_asset_id",
+            {"id": ASSET_ID},
+            "asset_id and asset_info cannot be provided simultaneously.",
+        ),
+    ],
+    ids=[
+        "Sc 1: Both asset_id and asset_info provided",
+        "Sc 2: Both asset_id and asset_info not provided",
+    ],
+)
+def test_init_asset_failure(auth_mock, asset_id, asset_info, expected_error):
+    with pytest.raises(ValueError) as err:
+        Asset(auth_mock, asset_id=asset_id, asset_info=asset_info)
+        assert expected_error == err
+
+
+def test_should_get_id_from_info(auth_mock, requests_mock):
+    url_asset_info = f"{auth_mock._endpoint()}/v2/assets/{ASSET_ID}/metadata"
+    requests_mock.get(url=url_asset_info, json=JSON_ASSET)
+    asset = Asset(auth=auth_mock, asset_info={"id": ASSET_ID})
+    assert asset.asset_id == ASSET_ID
+
+
```

Comment:
test1: test_should_initialize_with_provided_info - pass asset_info, unreachable auth_mock, verify info
test2: test_should_initialize_with_retrieved_info - pass asset_id, mock response data and verify the call and the result

Reply:
2 options same test with parametrize or 2 tests. I chose the first one. Please double-check any drawbacks.

Reply:
splitted